File: src/main/java/com/randomlychosenbytes/jlocker/nonabstractreps/RestrictedUser.java
package com.randomlychosenbytes.jlocker.nonabstractreps;

import com.randomlychosenbytes.jlocker.manager.Utils;

import javax.crypto.SecretKey;

public class RestrictedUser extends User {

    public RestrictedUser(String password, SecretKey key) {
        super(password);

        passwordHash = Utils.getHash(password); // MD5 hash
        encryptedUserMasterKeyBase64 = Utils.encryptKeyWithString(key, password);
    }
}

File: src/main/java/com/randomlychosenbytes/jlocker/model/RestrictedUser.kt
package com.randomlychosenbytes.jlocker.model

import com.randomlychosenbytes.jlocker.utils.encryptKeyWithString
import javax.crypto.SecretKey

class RestrictedUser(password: String, key: SecretKey) : User(
    password, encryptKeyWithString(key, password)
)